l4ping/l4ping.TESTGEN.ps1

```diff
@@ -0,0 +1,143 @@
+    Write-Output $cmd | Out-File -Encoding ascii -Append $g_logSend
+    Write-Output $cmd
+} # test_send()
+
+function test_l4ping_generate {
+    [CmdletBinding()]
+    Param(
+        [parameter(Mandatory=$true)] [String] $OutDir
+    )
+    # Normalize output directory
+    $dir = $OutDir
+
+    for ($i=0; $i -lt $g_Config.Iterations; $i++) {
+        # vary port number
+        [int] $port = $g_Config.StartPort + ($i * $g_Config.Iterations)
+
+        # Output receive command
+        test_recv -Port $port -ClientReceiveSize $g_Config.ClientReceiveSize -ClientSendSize $g_Config.ClientSendSize
+
+        # Output send command
+        test_send -Port $port -ClientReceiveSize $g_Config.ClientReceiveSize -ClientSendSize $g_Config.ClientSendSize -PingIterations $g_Config.PingIterations -Percentiles $g_Config.Percentiles -OutDir $dir -Fname "l4ping$Config.iter$i.csv"
+    }
+} # test_l4ping_generate()
+
+function validate_config {
+    $isValid = $true
+    $int_vars = @('Iterations', 'Port', 'ClientSendSize', 'ClientReceiveSize')
+    foreach ($var in $int_vars) {
+        if (($null -eq $g_Config.($var)) -or ($g_Config.($var) -lt 0)) {
+            Write-Output "$var is required and must be greater than or equal to 0"
+            $isValid = $false
+        }
+    }
+    return $isValid
+} # validate_config()
+
+#===============================================
+# External Functions - Main Program
+#===============================================
+function test_main {
+    Param(
+        [parameter(Mandatory=$false)] [string] $Config = "Default",
+        [parameter(Mandatory=$true)]  [string] $DestIp,
+        [parameter(Mandatory=$true)]  [string] $SrcIp,
+        [parameter(Mandatory=$true)]  [ValidateScript({Test-Path $_ -PathType Container})] [String] $OutDir = "" 
+    )
+    try {
+        # input_display
+        # get config variables
+        $allConfig = Get-Content -Path "$PSScriptRoot\l4ping.Config.json" | ConvertFrom-Json
+        [Object] $g_Config     = $allConfig.("l4ping$Config")
+        if ($null -eq $g_Config) {
+            Write-Output "l4ping$Config does not exist in .\l4ping\l4ping.Config.json. Please provide a valid config"
+            Throw
+        }
+        if (-Not (validate_config)) {
+            Write-Output "l4ping$Config is not a valid config"
+            Throw
+        }
+        [string] $g_DestIp     = $DestIp.Trim()
+        [string] $g_SrcIp      = $SrcIp.Trim()
+        [string] $dir          = (Join-Path -Path $OutDir -ChildPath "l4ping") 
+        [string] $g_log        = "$dir\l4ping.Commands.txt"
+        [string] $g_logSend    = "$dir\l4ping.Commands.Send.txt"
+        [string] $g_logRecv    = "$dir\l4ping.Commands.Recv.txt"
+
+        New-Item -ItemType directory -Path $dir | Out-Null
+        
+        # Optional - Edit spaces in output path for Invoke-Expression compatibility
+        # $dir  = $dir  -replace ' ','` '
+        banner -Msg "l4ping Tests"
```

Comment:
nit: single quotes

Reply:
Leaving as is based on offline discussion.